### PR TITLE
Add support for Ubuntu 20.04

### PR DESCRIPTION
**Pull request by ddeidda**
Packer script, recipes and configuration files needed to introduce the support for Ubuntu 20.04 on x86_64 and arm_64 architectures.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: files/ubuntu-20.04/configure_nw_interface.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+         via: 0.0.0.0
+         scope: link
+         table: ${ROUTE_TABLE}
+      routing-policy:
+        - from: ${DEVICE_IP_ADDRESS}
+          table: ${ROUTE_TABLE}
+EOF
+
+# Hook to delete automatic routes to subnet automatically created from kernel.
+# This is needed in Ubuntu 20 as well as for Ubuntu 18 for a slightly different reason. Automatic routes here are
```
Comment (rexcsn) [inline files/ubuntu-20.04/configure_nw_interface.sh:67]: nit: I think this file is almost identical to the one used for Ubuntu18. I see that there are comments here that explains the differences and why things are done in ubuntu20. Maybe we can just have 1 file with comments for both Ubuntu18 and 20?

File: files/ubuntu-20.04/ec2blkdev-init [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+# Required-Stop: $network
+# Default-Start: 2 3 4 5
+# Default-Stop: 0 1 6
+# Short-Description: ec2blkdev
+# Description: Script to call udevadm to setup any links to devices already attached at instance launch
+### END INIT INFO
+
+. /lib/lsb/init-functions
+
+prog=/bin/udevadm
```
Comment (rexcsn) [inline files/ubuntu-20.04/ec2blkdev-init:20]: I think this is the only line that is different between Ubuntu18 and Ubuntu20 version of the file? Maybe we can make this a template and substitute in the different programs, but only have 1 template for both ubuntu OSs?

File: files/ubuntu-20.04/sge_preinstall.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
```
Comment (rexcsn) [inline files/ubuntu-20.04/sge_preinstall.sh:1]: This file looks identical to the Ubuntu18 version. Can we use the same file under `files/ubuntu/` for both OSs? Same comment applies for other files that might be the same for Ubuntu OSs

File: files/ubuntu-20.04/configure_nw_interface.sh (hunk 1 above)
Comment (ddeidda) [reply files/ubuntu-20.04/configure_nw_interface.sh]: Agreed. I moved the files into a common `ubuntu` folder. 
Comments for `configure_nw_interfaces.sh` script have been merged, `ec2blkdev-init` logic unified.

File: files/ubuntu-20.04/ec2blkdev-init (hunk 2 above)
Comment (ddeidda) [reply files/ubuntu-20.04/ec2blkdev-init]: Done

File: files/ubuntu-20.04/sge_preinstall.sh (hunk 3 above)
Comment (ddeidda) [reply files/ubuntu-20.04/sge_preinstall.sh]: Done

**Review by rexcsn — DISMISSED** [review]

File: attributes/default.rb [hunk 4]
```diff
@@ -84,7 +84,7 @@
 default['cfncluster']['armpl']['platform'] = value_for_platform(
   'centos' => { '~>8' => 'RHEL-8' },
   'amazon' => { '2' => 'RHEL-8' },
-  'ubuntu' => { '18.04' => 'Ubuntu-16.04' }
+  'ubuntu' => { '>=18.04' => 'Ubuntu-16.04' }
```
Comment (lukeseawalker) [inline attributes/default.rb:87]: not related to this PR but why we are using `Ubuntu-16.04` archive for both 18.04 and 20.04?

File: recipes/setup_raid_on_head_node.rb [hunk 5]
```diff
@@ -59,6 +59,7 @@
   # Create RAID device with mdadm
   raid_superblock_version = value_for_platform(
     'centos' => { '>=8' => '1.2' },
+    'ubuntu' => { '>=20.04' => '1.2' },
```
Comment (lukeseawalker) [inline recipes/setup_raid_on_head_node.rb:62]: what is happening for U18 case? is it going to use the default value?

Comment (ddeidda) [reply recipes/setup_raid_on_head_node.rb:62]: Yes, 0.90 is used on Ubuntu 18

File: attributes/default.rb (hunk 4 above)
Comment (ddeidda) [reply attributes/default.rb:87]: It was the only available Ubuntu platform. Now I see that there is a new version, released on Mar 30, which should target Ubuntu 18.04. We might want to do the upgrade as a separate patch.

**Pull request merged**